src/main/app/src/components/common/CookieModal.js

```diff
@@ -0,0 +1,262 @@
+  const [cookiesAlreadyAccepted, setCookiesAlreadyAccepted] = useState(
+    Cookies.get(mandatoryCookieName)
+  );
+
+  const [statisticCookiesAccepted, setStatisticCookiesAccepted] = useState(false);
+  const [marketingCookiesAccepted, setMarketingCookiesAccepted] = useState(false);
+
+  const { data, isLoading } = useContentful();
+
+  const single = (entry) => Object.values(entry || [])[0] || {};
+
+  const contentfulTexts = single(data.cookieModalText);
+
+  const fields = {
+    shortContent: contentfulTexts['shortContent'] || '',
+    fullContent: contentfulTexts['fullContent'] || '',
+    heading: contentfulTexts['heading'] || '',
+    expandLinkText: contentfulTexts['expandLinkText'] || '',
+    settingsButtonText: contentfulTexts['settingsButtonText'] || '',
+    settingsButtonCloseText: contentfulTexts['settingsButtonCloseText'] || '',
+    acceptButtonText: contentfulTexts['acceptButtonText'] || 'Hyväksy',
+    settingsHeaderText: contentfulTexts['settingsHeaderText'] || '',
+    settingsAcceptMandatoryText: contentfulTexts['settingsAcceptMandatoryText'] || '',
+    settingsAcceptStatisticText: contentfulTexts['settingsAcceptStatisticText'] || '',
+    settingsAcceptMarketingText: contentfulTexts['settingsAcceptMarketingText'] || '',
+  };
+
+  function handleAcceptCookies(e) {
+    e.preventDefault();
+    Cookies.set(mandatoryCookieName, 'true', {
+      expires: 1800,
+      path: '/',
+    });
+    if (statisticCookiesAccepted) {
+      Cookies.set('oph-statistic-cookies-accepted', 'true', {
+        expires: 1800,
+        path: '/',
+      });
+    }
+    if (marketingCookiesAccepted) {
+      Cookies.set('oph-marketing-cookies-accepted', 'true', {
+        expires: 1800,
+        path: '/',
+      });
+    }
+    setCookiesAlreadyAccepted(true);
+  }
+
+  const openSettings = (
+    <div id="cookie-modal-settings" className={classes.settings}>
+      <hr />
+      <h3 className="cookie-modal-text"> {fields.settingsHeaderText}</h3>
+      <div>
+        <Checkbox
+          id="mandatoryCookies"
+          className={classes.settingsCheckbox}
+          checked
+          disabled
+        />
+        <label htmlFor="mandatoryCookies">{fields.settingsAcceptMandatoryText}</label>
+      </div>
+      <div>
+        <Checkbox
+          id="statisticCookies"
+          className={classes.settingsCheckbox}
+          checked={statisticCookiesAccepted}
+          disableripple
+          onClick={() => setStatisticCookiesAccepted(!statisticCookiesAccepted)}
+        />
+        <label htmlFor="statisticCookies">{fields.settingsAcceptStatisticText}</label>
+      </div>
+      <div>
+        <Checkbox
+          id="marketingCookies"
+          className={classes.settingsCheckbox}
+          checked={marketingCookiesAccepted}
+          disableripple
+          onClick={() => setMarketingCookiesAccepted(!marketingCookiesAccepted)}
+        />
+        <label htmlFor="marketingCookies">{fields.settingsAcceptMarketingText}</label>
```

Comment:
Checkboxille voi antaa childreninä suoraan labelin jolloin sen pitäs saada automaattisesti htmlFor
`<Checkbox ...>{checkboxinLabel}</Checkbox>`

Reply:
Tai ei taida olla htmlForia siinä mutta kuitenkin toimii interaktiivisena osana checkboxia, näin tehty hakutulossivulla. Pitäisi tyylit tulla myös suoraan oikein